src/tests/testClpMap.cc

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+#include "testClpMap.h"
+#include "unitTestMain.h"
+
+#include "SquidConfig.h"
+
+#include <ctime>
+
+
+CPPUNIT_TEST_SUITE_REGISTRATION( testClpMap );
+
+class SquidConfig Config;
+
+void
+testClpMap::addData(testMap &m, int numElems, int base, testMap::Ttl ttl)
+{
+    for (int j = base; j < base + numElems; ++j ) {
+        CPPUNIT_ASSERT_EQUAL(true, m.add(std::to_string(j), j, ttl));
+    }
+}
+
+
+void
+testClpMap::setUp()
+{
+    squid_curtime = time(nullptr);
+}
+
+void
+testClpMap::PutGetDelete()
+{
+    testMap m(1024);
+    addData(m, 10);
+    CPPUNIT_ASSERT_EQUAL(static_cast<const int *>(nullptr), m.get("notthere"));
```

Comment:
CPPUNIT_ASSERT_EQUAL() is best when the printed values tell us something. In this and similar "not found" cases, the printed values are meaningless pointers. I think this would be a lot more readable:

```suggestion
    CPPUNIT_ASSERT(!m.get("notthere"));
```

Reply:
ok